File: code_samples/go/design_patterns_adapter.go
package main

import (
    "fmt"
)

// Design Patterns: Adapter
// AI/ML Training Sample

type Adapter struct {
    Data string
}

func NewAdapter() *Adapter {
    return &Adapter{
        Data: "",
    }
}

func (s *Adapter) Process(input string) {
    s.Data = input
}

func (s *Adapter) Validate() bool {
    return len(s.Data) > 0
}

func (s *Adapter) GetData() string {
    return s.Data
}

func main() {
    instance := NewAdapter()
    instance.Process("example")
    fmt.Printf("Data: %s\n", instance.GetData())
    fmt.Printf("Valid: %v\n", instance.Validate())
}

File: code_samples/haskell/design_patterns_adapter.hs
-- Design Patterns: Adapter
-- AI/ML Training Sample

module Adapter where

data Adapter = Adapter {
    getData :: String
} deriving (Show, Eq)

process :: Adapter -> String -> Adapter
process obj input = obj { getData = input }

validate :: Adapter -> Bool
validate obj = not (null (getData obj))

main :: IO ()
main = do
    let instance = Adapter { getData = "" }
    let updated = process instance "example"
    putStrLn $ "Data: " ++ getData updated
    putStrLn $ "Valid: " ++ show (validate updated)
